packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignInButtons/__tests__/FederatedSignInButton.test.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+import { FederatedIdentityProviders } from '@aws-amplify/ui';
+import * as ReactCoreModule from '@aws-amplify/ui-react-core';
+
+import { FederatedSignInButton } from '../FederatedSignInButton';
+
+const SUPPORTED_PROVIDERS = [
+  FederatedIdentityProviders.Amazon,
+  FederatedIdentityProviders.Apple,
+  FederatedIdentityProviders.Facebook,
+  FederatedIdentityProviders.Google,
+];
+
+const useAuthenticatorMock = {
+  toFederatedSignIn: jest.fn(),
+};
+
+jest
+  .spyOn(ReactCoreModule, 'useAuthenticator')
+  .mockImplementation(() => useAuthenticatorMock as any);
+
+describe('FederatedSignInButton', () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
+  });
+
+  it.each(SUPPORTED_PROVIDERS)(
```

Comment:
Unrelated to #2697, just added this while I was here.

Reply:
💯